src/storage/src/manifest/storage.rs

```diff
@@ -321,6 +321,40 @@ impl ManifestLogStorage for ManifestObjectStore {
         Ok(ret)
     }
 
+    async fn delete_all(&self) -> Result<()> {
+        let mut entries: Vec<(ManifestVersion, Entry)> = self
+            .get_paths(|entry| {
+                let file_name = entry.name();
+                if is_delta_file(file_name) {
+                    let version = file_version(file_name);
+                    return Some((version, entry));
+                }
+                None
+            })
+            .await?;
+
+        entries.sort_unstable_by(|(v1, _), (v2, _)| v1.cmp(v2));
+        let paths: Vec<_> = entries.iter().map(|e| e.1.path().to_string()).collect();
+
+        // Delete logs in order of verseion.
+        self.object_store
+            .remove(paths)
+            .await
+            .with_context(|_| DeleteObjectSnafu {
+                path: self.path.clone(),
+            })?;
```

Comment:
Looks like `remove_all` is enough, it will delete all files and directory recusively.
```suggestion
```

Reply:
Some files are left under the manifest directory if failures occur during executing `remove_all` (e.g. restart, network failure). It'd be better to delete the json file that contains the `Remove` meta action after other files are deleted.